From 458d48aee50171d3dec0ad6c9502bf7d795f9da1 Mon Sep 17 00:00:00 2001
From: Stanislav Levin <slev@altlinux.org>
Date: Mon, 28 Feb 2022 13:12:37 +0300
Subject: [PATCH] tests: Sync expected stream for Pytest's version

https://docs.pytest.org/en/7.0.x/changelog.html#breaking-changes:
> [pytest#8246](https://github.com/pytest-dev/pytest/issues/8246): --version now writes version information to stdout rather than stderr.

Fixes: https://github.com/pytest-dev/pytest-django/issues/995
Signed-off-by: Stanislav Levin <slev@altlinux.org>
---
 tests/test_manage_py_scan.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/tests/test_manage_py_scan.py b/tests/test_manage_py_scan.py
index 39544589..490882b0 100644
--- a/tests/test_manage_py_scan.py
+++ b/tests/test_manage_py_scan.py
@@ -118,7 +118,12 @@ def test_django_project_found_invalid_settings_version(django_testdir, monkeypat
 
     result = django_testdir.runpytest_subprocess("django_project_root", "--version", "--version")
     assert result.ret == 0
-    result.stderr.fnmatch_lines(["*This is pytest version*"])
+    if hasattr(pytest, "version_tuple") and pytest.version_tuple >= (7, 0):
+        version_out = result.stdout
+    else:
+        version_out = result.stderr
+
+    version_out.fnmatch_lines(["*This is pytest version*"])
 
     result = django_testdir.runpytest_subprocess("django_project_root", "--help")
     assert result.ret == 0
